src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryLayout.java

```diff
@@ -350,16 +348,36 @@ default long byteAlignment() {
      * in {@code elements} is {@code null}.
      */
     default long bitOffset(PathElement... elements) {
-        return computePathOp(LayoutPath.rootPath(this, MemoryLayout::bitSize), LayoutPath::offset, EnumSet.of(PathKind.SEQUENCE_ELEMENT, PathKind.SEQUENCE_RANGE), elements);
+        return computePathOp(LayoutPath.rootPath(this, MemoryLayout::bitSize), LayoutPath::offset,
+                EnumSet.of(PathKind.SEQUENCE_ELEMENT, PathKind.SEQUENCE_RANGE), elements);
+    }
+
+    /**
+     * Creates a method handle that can be used to compute the offset, in bits, of the layout selected
+     * by a given layout path, where the path is considered rooted in this layout.
+     *
+     * <p>This is like {@link #bitOffset(PathElement...)}, except that free dimensions
```

Comment:
I'd replace this text, or maybe move it after we explain what the signature of the returned method handle is. E.g.

```
The returned method handle has a return type of `long`, and features as many `long` parameter types as there are free dimensions in the provided layout path. In other words, this method can be used as a replacements in cases where {@link #bitOffset(PathElement...)} does not work.
```
Or something similar.
```